tools/github-action-reporter.js

```diff
@@ -0,0 +1,79 @@
+module.exports = {
+  "reporter:respec-github-action": ["type", GithubActionReporter],
+};
+
+function GithubActionReporter() {
+  const browserFailures = [];
+  this.onSpecComplete = (_browser, result) => {
+    if (!result.success) {
+      const failure = parseFailure(result);
+      browserFailures.push(failure);
+    }
+  };
+
+  this.onRunComplete = () => {
+    print("group", "Failed tests");
+    browserFailures.forEach(printFailure);
+    print("endgroup");
+  };
+}
+
+/**
+ * @typedef {{ test: string[], file: string, line: number, col: number, message: string }} Failure
+ * @returns {Failure}
+ */
+function parseFailure(result) {
+  // convert newlines into array and flatten
+  const log = result.log.flatMap(message => message.split("\n"));
+  const { suite, description } = result;
+  const message = log[0].replace("Error: ", "");
+  const location = log[2].split(":9876/base/", 2)[1].replace(/\)$/, "");
+  // eslint-disable-next-line prefer-const
+  let [file, line, col] = location.split(":");
+  line = parseInt(line, 10);
+  col = parseInt(col, 10);
+  const test = suite.concat(description);
+  return { test, file, line, col, message };
+}
+
+/** @param {Failure} failure */
+function printFailure(failure) {
+  const { file, line, col, test, message } = failure;
+  const msg = test
+    .map((s, i) => `${" ".repeat(i * 2)}${s}`)
+    .concat([message])
+    .join("\n");
+  print("error", msg, { file, line, col });
+}
+
+/**
+ * @param {string} command
+ * @param {string} message
+ * @param {Record<string, string>} options
+ */
+function print(command, message = "", options = {}) {
+  let optionsString = Object.entries(options)
+    .map(([k, v]) => `${k}=${escapeProperty(v)}`)
```

Comment:
Node has `URLSearchParams` too, you can just use it instead of escaping everything manually.

Reply:
Let me try... I think this is using a different escaping than URLSearchParams. I copied it from [@actions/core](https://github.com/actions/toolkit/tree/master/packages/core)

Reply:
Didn't work :disappointed: 
Did I do what you expected or you meant something else?

Reply:
URLSearchParams uses `&` as the separator and here it's `,`, so `.replace(/&/, ",")` is needed.